m4/odp_atomic.m4

```diff
@@ -126,12 +126,13 @@ AC_CACHE_CHECK([whether -latomic is needed for 64-bit atomic compare exchange],
 	       [odp_cv_atomic_needed_64bit_cmp_exc], [dnl
 AC_LINK_IFELSE(
   [AC_LANG_SOURCE([[
+    #include <stdbool.h>
     #include <stdint.h>
     static uint64_t loc;
     int main(void)
     {
         uint64_t exp = 0;
-        uint64_t = __atomic_compare_exchange_n(&loc, &exp, 1, 1,
+        bool res = __atomic_compare_exchange_n(&loc, &exp, 1, 1,
```

Comment:
Note that this was broken from the initial commit of this code. That is this code was never compiling (because it had invalid syntax). After fixing the syntax it seems clang linking fails on `odp_ipfragreass_atomics.h`. Naturally changing this back to effectively force `-latomic` makes the compilation pass but that is not the correct approach.